### PR TITLE
Fix http2 support on 2.4 in debian by using newer OpenSSL 1.0.2+

### LINKED ISSUE #42: ALPN Not supported
I cannot get HTTP2 support working off of `httpd:2.4`, despite enabling the mods and adding `Protocol` directives in my `VirtualHost`s. I think it's because the OpenSSL that's being used is too old. The `debian:jessie-backports` includes OpenSSL 1.0.2, which works with ALPN and thus HTTP2; any chance you'll be upgrading the packages?

Thanks!

### PR DESCRIPTION
**Pull request by yosifkit**
Fixes #42.

```console
$ curl -v --http2 https://server-with-h2-and-ssl
...
* ALPN, offering h2
* ALPN, offering http/1.1
* SSL connection using TLS1.2 / ECDHE_RSA_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
```

### CODE REVIEW

File: update.sh [hunk 1]
```diff
@@ -10,6 +10,7 @@ fi
 versions=( "${versions[@]%/}" )
 
 nghttp2VersionDebian="$(docker run -i --rm debian:stretch-slim bash -c 'apt-get update -qq && apt-cache show "$@"' -- "libnghttp2-dev" |tac|tac| awk -F ': ' '$1 == "Version" { print $2; exit }')"
+opensslVersionDebian="$(docker run -i --rm debian:jessie-backports bash -c 'apt-get update -qq && apt-cache show "$@"' -- "openssl" |tac|tac| awk -F ': ' '$1 == "Version" { print $2; exit }')"
```
Comment (tianon) [inline update.sh]: This is potentially dangerous:
```console
$ docker run -i --rm debian:jessie-backports bash -c 'apt-get update -qq && apt-cache show "$@"' -- "openssl"
Package: openssl
Version: 1.0.2j-1~bpo8+1
Installed-Size: 1135
Maintainer: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Architecture: amd64
Depends: libc6 (>= 2.15), libssl1.0.0 (>= 1.0.2~beta3)
Suggests: ca-certificates
Description: Secure Sockets Layer toolkit - cryptographic utility
Description-md5: 9b6de2bb6e1d9016aeb0f00bcf6617bd
Section: utils
Priority: optional
Filename: pool/main/o/openssl/openssl_1.0.2j-1~bpo8+1_amd64.deb
Size: 686206
MD5sum: f2e1ea656353992ad62d4e984c7454cd
SHA1: 3cfc4d4460c33bf8b3cdb7deec3274dee2184a7c
SHA256: ab15b3b09c296ceac92763e8163f8a1337bf8e4a866d51706442f1c0d501498f

Package: openssl
Version: 1.0.1t-1+deb8u5
Installed-Size: 1093
Maintainer: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Architecture: amd64
Depends: libc6 (>= 2.15), libssl1.0.0 (>= 1.0.1k-3+deb8u3)
Suggests: ca-certificates
Description: Secure Sockets Layer toolkit - cryptographic utility
Description-md5: 9b6de2bb6e1d9016aeb0f00bcf6617bd
Section: utils
Priority: optional
Filename: pool/updates/main/o/openssl/openssl_1.0.1t-1+deb8u5_amd64.deb
Size: 664872
MD5sum: 3796988cdc8c65ac9516012274d04ec0
SHA1: 8493eb197d20ec1aceeb2a44139b6efbc2b4323b
SHA256: a70d960a12802daf27e86655d936aedeef52dd030809d6ba4cb0a1ee7f7097e5

Package: openssl
Version: 1.0.1t-1+deb8u3
Installed-Size: 1092
Maintainer: Debian OpenSSL Team <pkg-openssl-devel@lists.alioth.debian.org>
Architecture: amd64
Depends: libc6 (>= 2.15), libssl1.0.0 (>= 1.0.1k-3+deb8u3)
Suggests: ca-certificates
Description: Secure Sockets Layer toolkit - cryptographic utility
Description-md5: 9b6de2bb6e1d9016aeb0f00bcf6617bd
Tag: implemented-in::c, interface::commandline, protocol::ssl, role::program,
 scope::utility, security::cryptography, security::integrity,
 use::checking
Section: utils
Priority: optional
Filename: pool/main/o/openssl/openssl_1.0.1t-1+deb8u3_amd64.deb
Size: 664334
MD5sum: 75adfa928f299f3f3d1d6c2a73e13654
SHA1: f87a4ef9d51e2e0f57758342e0433cc44129c6ca
SHA256: 9d04e590361bc6b14e71940f20f61fe60a18d0e3e8963c50c321506e53b288ee

```

Comment (tianon) [inline update.sh]: That does appear to be ordered by version number descending, so guess we might as well try our luck and figure out a better fix if/when it causes a problem. :+1:

Comment (yosifkit) [inline update.sh]: I'm fine with "fix it when it breaks".  (apologies to my future self :disappointed: :bowing_man:)

**Comment by koehn** [comment]
I approve! For whatever that's worth. 

Seriously, it would be great to have this.

**Comment by koehn** [comment]
Thanks so much! When should we expect to see it in Docker Hub?

**Comment by tianon** [comment]
@koehn https://github.com/docker-library/official-images/pull/2539 (which just merged, so should be built and pushed shortly) :+1:

**Pull request merged**